pkg/kritis/attestation/attestation.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package attestation defines methods to attest a message using Pgp Private and
+// Public Key pair.
+package attestation
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/grafeas/kritis/pkg/kritis/admission/constants"
+	"github.com/pkg/errors"
+
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/packet"
+	"golang.org/x/crypto/openpgp/s2k"
+)
+
+var pgpConfig = packet.Config{
+	// Use Sha256
+	DefaultHash:            crypto.SHA256,
+	DefaultCipher:          packet.CipherAES256,
+	DefaultCompressionAlgo: packet.CompressionZLIB,
+	CompressionConfig: &packet.CompressionConfig{
+		Level: packet.DefaultCompression,
+	},
+	RSABits: constants.RSABits,
+}
+
+// VerifyImageAttestation verifies if the image is attested using the Base64
+// encoded public key.
+func VerifyImageAttestation(pubKeyEnc string, attestationHash string, message string) error {
+	pemPublicKey, err := base64.StdEncoding.DecodeString(pubKeyEnc)
+	if err != nil {
+		return err
+	}
+	attestation, err := base64.StdEncoding.DecodeString(attestationHash)
+	if err != nil {
+		return err
+	}
+
+	keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(string(pemPublicKey)))
+	if err != nil {
+		return err
+	}
+	buf := bytes.NewBuffer([]byte(attestation))
+	armorBlock, err := armor.Decode(buf)
+	md, err := openpgp.ReadMessage(armorBlock.Body, keyring, nil, &pgpConfig)
+
+	// Verify Signature using the Public Key
+	plaintext, err := ioutil.ReadAll(md.UnverifiedBody)
+	if err != nil {
+		return err
+	}
+
+	if string(plaintext) != message {
+		return fmt.Errorf("Signature could not be verified. Got: %q, Want: %q", plaintext, message)
+	}
+	return nil
+}
+
+// AttestMessage attests the message using the given public and private key.
+// pubKeyEnc: Base64 Encoded Public Key
+// privKeyEnc: Base64 Decoded Private Key
+// message: Message to attest
+func AttestMessage(pubKeyEnc string, privKeyEnc string, message string) (string, error) {
```

Comment:
Nit/Optional: I would probably name this CreateImageAttestation to be aligned with VerifyImageAttestation.  Or name the other one also "VerifyMessageAttestation" as there really is no image specific code in here.

Could you point me to the code where this is called?

Reply:
it should be called here https://github.com/grafeas/kritis/blob/master/pkg/kritis/admission/admission.go#L108 where the todo is mentioned. 
We will first build a message of type  Red Hat’s Atomic Host attestation signature

Reply:
Added more details TODO here https://github.com/grafeas/kritis/pull/52/commits/b8cdbe9f336bf43c83ed2b3c833928b0895b7598